code/modules/projectiles/guns/projectile.dm

```diff
@@ -77,18 +78,22 @@
 				for(var/i = 1; i<=min(to_drop, stored_magazine.stored_ammo.len); i++)
 					var/obj/item/ammo_casing/AC = stored_magazine.stored_ammo[1]
 					stored_magazine.stored_ammo -= AC
-					AC.forceMove(get_turf(user))
+					AC.forceMove(user.loc)
 					dropped_bullets++
 					stored_magazine.update_icon()
 				to_chat(usr, "<span class='notice'>You unjam the [name], and spill [dropped_bullets] bullet\s in the process.</span>")
 				chamber_round()
 				update_icon()
 				return 0
 			return 0
-		stored_magazine.forceMove(get_turf(src.loc))
+		stored_magazine.forceMove(get_turf(src.loc)) //this first drops the magazine onto the turf, it's here in case there is no applicable user
```

Comment:
for the same reason you're moving the spent casing to the loc instead of the floor, you should probably do the same with the mag

Reply:
kur, read into it more, it places the mag in the user.loc later, this here is mostly set against the next user check in case the user is unavailable